sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1906,291 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+class QuantileNormalizer(BaseEstimator, TransformerMixin):
+    """Normalize features using quantiles information.
+
+    This Normalizer scales the features between 0 and 1, equalizing the
+    distribution of each feature to a uniform distribution. Therefore,
+    for a given feature, this normalization tends to spread out the most
+    frequent values.
+
+    The normalization is applied on each feature independently.
+    The cumulative density function of a feature is used to project the
+    original values.
```

Comment:
Add something like:

Features of new/unseen data that fall below or above the fitted range will be mapped to 0 and one, respectively.
Note that this transform and non-linear. It may remove correlations between variables measured at the same scale but renders variables measured at different scales more directly comparable.